lib/openshift/prometheus.rb

```diff
@@ -0,0 +1,12 @@
+require 'openshift/project_resource'
+
+module BushSlicer
+  class Prometheus < ProjectResource
+    RESOURCE = "prometheus"
```

Comment:
could you set the full qualified API here? That is with `.openshift.io` or whatever it is for this resource.

Reply:
`oc get prometheus -n openshift-monitoring -oyaml
apiVersion: v1
items:
- apiVersion: monitoring.coreos.com/v1
  kind: Prometheus
  metadata:
    creationTimestamp: "2020-08-19T06:29:20Z"
    generation: 1
    labels:
      prometheus: k8s
    managedFields:
    - apiVersion: monitoring.coreos.com/v1
`
I change the line as
`
  RESOURCE = "prometheus,monitoring.coreos.com"
`
Get the following error:
`waiting for operation up to 3600 seconds..
    And I wait for the "user-workload" prometheus to appear up to 120 seconds                                                                       # features/step_definitions/resources.rb:144
      hash not from a Prometheus: expected  but was Prometheus (RuntimeError)
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:140:in `update_from_api_object'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:82:in `get'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:43:in `exists?'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:201:in `block in wait_to_appear'
      /home/hongyli/openshift/projects/verification-tests/lib/base_helper.rb:153:in `wait_for'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:200:in `wait_to_appear'
      /home/hongyli/openshift/projects/verification-tests/features/step_definitions/resources.rb:149:in `/^(I|admin) waits? for the "(.+?)" (\w+) to appear(?: in the(?: "(.+?)")? project)?(?: up to (\d+) seconds)?$/'
      ./features/tierN/monitoring/ci.feature:1195:in `And I wait for the "user-workload" prometheus to appear up to 120 seconds'
    And I wait for the "user-workload" thanos_ruler to appear up to 120 seconds                                                                     # features/step_definitions/resources.rb:144
      #set log level`

Reply:
Could you paste full YAML output? The `prometheus` part may need plural but it is not apparent from current output.

wrt the error, are you using latest repo code? I think such error was fixed some months ago. See lib/openshift/etcd_cluster.rb for example.

Reply:
Actually looked at the code and the error perhaps comes exactly because "prometheus.monitoring.coreos.com" is incorrect. Please paste full YAML and I will be able to tell (or will be able to reproduce).

Reply:
Full YAML
http://pastebin.test.redhat.com/897376

Reply:
Cool, so look at `selfLink`, it is `prometheuses`. Thus `RESOURCE = "prometheuses.monitoring.coreos.com"`. You can check `selfLink` for the other resources also.

Reply:
Fixed an issue with f960862 so please rebase before you continue.